app/actions/views/permalink.ts

```diff
@@ -2,44 +2,110 @@
+                        if (channelError) {
+                            return {error: channelError};
+                        }
+                        if (channelData.type === General.PRIVATE_CHANNEL) {
+                            Alert.alert(
+                                intl.formatMessage({
+                                    id: 'permalink.show_dialog_warn.title',
+                                    defaultMessage: 'Private Channel',
+                                }),
+                                intl.formatMessage({
+                                    id: 'permalink.show_dialog_warn.description',
+                                    defaultMessage: 'You must join "{channel}" to view this post.',
+                                }, {
+                                    channel: channelData.name,
+                                }),
+                                [
+                                    {
+                                        text: intl.formatMessage({
+                                            id: 'permalink.show_dialog_warn.cancel',
+                                            defaultMessage: 'Cancel',
+                                        }),
+                                    },
+                                    {
+                                        text: intl.formatMessage({
+                                            id: 'permalink.show_dialog_warn.join',
+                                            defaultMessage: 'Join',
+                                        }),
+                                        onPress: async () => {
+                                            await dispatch(joinChannel(currentUserId, teamName, channelId, channelData.name));
+                                            dispatch(processShowPermaLink(postId, openAsPermalink));
+                                        },
+                                    },
+                                ],
+                            );
+                            return null;
+                        }
+                    }
+                }
             }
+            dispatch(processShowPermaLink(postId, openAsPermalink));
+        }
+        return null;
+    };
+}
+
+function processShowPermaLink(postId: string, openAsPermalink: boolean) {
+    return async (dispatch: DispatchFunc) => {
+        dispatch(selectFocusedPostId(postId));
+        if (!showingPermalink) {
+            const screen = 'Permalink';
+            const passProps = {
+                isPermalink: openAsPermalink,
+                onClose: () => {
+                    dispatch(closePermalink());
+                },
+            };
+
+            const options = {
+                layout: {
+                    componentBackgroundColor: changeOpacity('#000', 0.2),
+                },
+            };
+
+            showingPermalink = true;
+            showModalOverCurrentContext(screen, passProps, options);
         }
+        return {};
```

Comment:
Do we have to return an empty object? what happens if we dont?

Reply:
Typescript error would be shown as we are doing **dispatch** and it expects an object to be  returned.

Reply:
not an empty object  though, it should have data or error properties.

Reply:
cool

Reply:
Which is true but in this case, return value isn't expected from this function. Now I have added an empty data object to make return value consistent.